listenbrainz/db/feedback.py

```diff
@@ -0,0 +1,115 @@
+import sqlalchemy
+
+from listenbrainz import db
+from listenbrainz.feedback import Feedback
+
+
+def insert(feedback: Feedback):
+    """ Inserts a feedback record for a user's loved/hated recording into the database.
+        If the record is already present for the user, the score is updated to the new
+        value passed.
+        Args:
+            feedback: An object of class Feedback
+    """
+
+    with db.engine.connect() as connection:
+        connection.execute(sqlalchemy.text("""
+            INSERT INTO recording_feedback (user_id, recording_msid, score)
+                 VALUES (:user_id, :recording_msid, :score)
+            ON CONFLICT (user_id, recording_msid)
+          DO UPDATE SET score = :score,
+                        created = NOW()
+            """), {
+                'user_id': feedback.user_id,
+                'recording_msid': feedback.recording_msid,
+                'score': feedback.score,
+            }
+        )
+
+
+def delete(feedback: Feedback):
+    """ Deletes the feedback record for a given recording for the user from the database.
+        Args:
+            feedback: An object of class Feedback
+    """
+
+    with db.engine.connect() as connection:
+        connection.execute(sqlalchemy.text("""
+            DELETE FROM recording_feedback
+             WHERE user_id = :user_id
+               AND recording_msid = :recording_msid
+            """), {
+                'user_id': feedback.user_id,
+                'recording_msid': feedback.recording_msid,
+            }
+        )
+
+
+def get_feedback_by_user_id(user_id: int):
+    """ Get a list of recording feedbacks given by the user.
+        Args:
+            user_id: the row ID of the user in the DB
+        Returns:
+            A list of user's feedback
+    """
+    records = []
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT user_id, recording_msid, score
+              FROM recording_feedback
+          ORDER BY created DESC
+            """), {
+                'user_id': user_id
+            }
+        )
+        return [Feedback(user_id=row["user_id"], recording_msid=str(row["recording_msid"]), score=row["score"])
+                for row in result.fetchall() if row['user_id'] is not None]
```

Comment:
the definition of the user_id field in the database schema is `NOT NULL`, so this will never be None. Therefore I don't think the `if` is needed.

Reply:
Yep